### ISSUE TITLE
Android Debugger falls into infinity loop after a build.

### ISSUE THREAD

**Issue by kirillta** [Microsoft/vscode-react-native#464]
## Actual Behavior
1. I press DEBUG button with "Debug Android" option.
2. Android Debugger falls into infinity loop few seconds after a build completion. The app works with `react-native run-android`, but it couldn't establish a connection with an emulator in four cases out of five.

![debugger-loop](https://user-images.githubusercontent.com/19779590/27799224-0807761e-601d-11e7-85de-bd7595fd36fb.gif)

## Expected Behavior
Debugger stops on an exception.

## Software versions
- React-Native VS Code extension version: 0.3.2
- VSCode version: 1.14.0-insider (2017-07-03T11:15:12.102Z)
- OS platform and version: win 10 x64
- NodeJS version: 7.4.0
- React Native version: 0.40

## Outputs (Include if relevant)
- Output of the Debug Console (View -> Toggle Debug Console):
```
[vscode-react-native] Starting React Native Packager.
[vscode-react-native] Prewarming bundle cache. This may take a while ...
[vscode-react-native] Building and running application.
[vscode-react-native] Executing command: react-native.cmd run-android
JS server already running.
Running c:\Program Files (x86)\Android\android-sdk\/platform-tools/adb -s emulator-5554 reverse tcp:8081 tcp:8081
Building and installing the app on the device (cd android && gradlew.bat installDebug)...
:app:preBuild UP-TO-DATE
:app:preDebugBuild UP-TO-DATE
```
I omit the rest of checks here.
```
Installing APK 'app-debug.apk' on 'emulator-5554 - 6.0'
Installed on 1 device.

BUILD SUCCESSFUL

Total time: 22.91 secs
Starting the app on emulator-5554 (c:\Program Files (x86)\Android\android-sdk\/platform-tools/adb -s emulator-5554 shell am start -n com.barfriends/.MainActivity)...
Starting: Intent { cmp=com.barfriends/.MainActivity }

[vscode-react-native] Finished executing: react-native.cmd run-android
[vscode-react-native] Executing command: adb -s emulator-5554 shell am broadcast -a "com.barfriends.RELOAD_APP_ACTION" --ez jsproxy true
[vscode-react-native] Broadcasting: Intent { act=com.barfriends.RELOAD_APP_ACTION (has extras) }

Broadcast completed: result=0


[vscode-react-native] Finished executing: adb -s emulator-5554 shell am broadcast -a "com.barfriends.RELOAD_APP_ACTION" --ez jsproxy true
[vscode-react-native] Starting debugger app worker.
[vscode-react-native] Established a connection with the Proxy (Packager) to the React Native application
[vscode-react-native] Debugger worker loaded runtime on port 3887
[vscode-react-native] Debugger worker loaded runtime on port 5442
[vscode-react-native] Starting React Native Packager.
[vscode-react-native] Prewarming bundle cache. This may take a while ...
[vscode-react-native] Building and running application.
[vscode-react-native] Executing command: react-native.cmd run-android
JS server already running.
Running c:\Program Files (x86)\Android\android-sdk\/platform-tools/adb -s emulator-5554 reverse tcp:8081 tcp:8081
Building and installing the app on the device (cd android && gradlew.bat installDebug)...
:app:preBuild UP-TO-DATE
:app:preDebugBuild UP-TO-DATE
```
I omit the rest of checks here.
```
Installing APK 'app-debug.apk' on 'emulator-5554 - 6.0'
Installed on 1 device.

BUILD SUCCESSFUL

Total time: 22.962 secs
Starting the app on emulator-5554 (c:\Program Files (x86)\Android\android-sdk\/platform-tools/adb -s emulator-5554 shell am start -n com.barfriends/.MainActivity)...
Starting: Intent { cmp=com.barfriends/.MainActivity }

[vscode-react-native] Finished executing: react-native.cmd run-android
[vscode-react-native] Executing command: adb -s emulator-5554 shell am broadcast -a "com.barfriends.RELOAD_APP_ACTION" --ez jsproxy true
```

and so on...

**Comment by iainroyle** [comment]
I can confirm the same thing. White screen every time I started the debugger except for the very first time when it worked for some reason.

Works with RN 0.44 it seems.

**Comment by lyingd** [comment]
+1
RN: 0.44
vscode: 1.14.0

**Comment by vladimir-kotikov** [comment]
@kirillta, @iainroyle, @lyingd - I guess this issue has the same reason as #470 and #471 - this is confirmed indirectly that all you're using VSCode 1.14 (either stable or insiders)

**Comment by vladimir-kotikov** [comment]
This has been fixed by #472, closing.
